diff -Naur grpc-1.26.0-original/src/python/grpcio_tests/setup.py grpc-1.26.0/src/python/grpcio_tests/setup.py
--- grpc-1.26.0-original/src/python/grpcio_tests/setup.py	2019-12-17 20:29:27.000000000 -0500
+++ grpc-1.26.0/src/python/grpcio_tests/setup.py	2021-02-02 09:11:20.236194502 -0500
@@ -37,7 +37,7 @@
 }
 
 INSTALL_REQUIRES = (
-    'coverage>=4.0', 'enum34>=1.0.4',
+    'coverage>=4.0',
     'grpcio>={version}'.format(version=grpc_version.VERSION),
     'grpcio-channelz>={version}'.format(version=grpc_version.VERSION),
     'grpcio-status>={version}'.format(version=grpc_version.VERSION),
@@ -48,6 +48,8 @@
 
 if not PY3:
     INSTALL_REQUIRES += ('futures>=2.2.0',)
+if sys.version_info < (3, 4):
+    INSTALL_REQUIRES += ('enum34>=1.0.4',)
 
 COMMAND_CLASS = {
     # Run `preprocess` *before* doing any packaging!
