Backport upstream commit 9e0b427893b65b220faf8a31a6afdc67f6f41364 “Use != with literals”

diff -Naur grpc-1.26.0-original/src/python/grpcio/commands.py grpc-1.26.0/src/python/grpcio/commands.py
--- grpc-1.26.0-original/src/python/grpcio/commands.py	2019-12-17 20:29:27.000000000 -0500
+++ grpc-1.26.0/src/python/grpcio/commands.py	2021-01-28 12:51:33.635142522 -0500
@@ -99,7 +99,7 @@
         target_dir = os.path.join(GRPC_STEM, 'doc', 'build')
         exit_code = sphinx.cmd.build.build_main(
             ['-b', 'html', '-W', '--keep-going', source_dir, target_dir])
-        if exit_code is not 0:
+        if exit_code:
             raise CommandError(
                 "Documentation generation has warnings or errors")
 
